satpy/tests/reader_tests/test_hsaf_h5.py

```diff
@@ -0,0 +1,67 @@
+"""Tests for the H-SAF H5 reader."""
+import os
+from datetime import datetime
+
+import h5py
+import numpy as np
+import pytest
+
+from satpy import Scene
+from satpy.resample import get_area_def
+
+# real shape is 916, 1902
+shape_sc = (916, 1902)
+shape_sc_colormap = (256, 3)
+values_sc = np.random.randint(0, 9, shape_sc, dtype=np.uint8)
+
+dataset_names = {"SC": "Snow Cover",
+                 "SC_pal": "Snow Cover Palette"}
+
+dimensions = {"SC": shape_sc,
+              "colormap": shape_sc_colormap, }
+
+start_time = datetime(2022, 11, 15, 0, 0)
+end_time = datetime(2022, 11, 15, 0, 0)
+
+
+@pytest.fixture(scope="session")
+def hsaf_filename():
+    """Create a fake HSAF SC HDF5 file."""
+    filename = "h10_20221115_day_merged.H5"
+    with h5py.File(filename, mode="w") as h5f:
+        h5f.create_dataset('SC', shape_sc, dtype=np.uint8)
+        h5f.create_dataset('colormap', shape_sc_colormap, dtype=np.uint8)
+    return filename
+
+
+def test_hsaf_sc_dataset(hsaf_filename):
+    """Test the H-SAF SC dataset."""
+    scn = Scene(filenames=[str(hsaf_filename)], reader="hsaf_h5")
+    scn.load(['SC'])
+    assert scn['SC'].shape == shape_sc
+
+
+def test_hsaf_sc_colormap_dataset(hsaf_filename):
+    """Test the H-SAF SC_pal dataset."""
+    scn = Scene(filenames=[str(hsaf_filename)], reader="hsaf_h5")
+    scn.load(['SC_pal'])
+    assert scn['SC_pal'].shape == shape_sc_colormap
+
+
+def test_hsaf_sc_datetime(hsaf_filename):
+    """Test the H-SAF reference time."""
+    scn = Scene(filenames=[str(hsaf_filename)], reader="hsaf_h5")
+    scn.load(['SC'])
+    fname = str(os.path.basename(hsaf_filename))
+    dtstr = fname.split('_')[1].zfill(4)
+    obs_time = datetime.strptime(dtstr, "%Y%m%d%H%M")
```

Comment:
Just a initial comment to fix the test. The original test is parsinga a different datetime (2022-01-01 01:05) compared to the filename (2022-11-15). So also the reader is returning a wrong time as the tests pass..
```suggestion
    fname = os.path.basename(hsaf_filename)
    dtstr = fname.split('_')[1]
    obs_time = datetime.strptime(dtstr, "%Y%m%d")
```